### DISCUSSION TITLE
Trigger with Mouse/Trackpad Gestures via BTT / Keyboard Shortcut

### DISCUSSION THREAD

**Discussion by hairybridget** [rxhanson/Charmstone-Community#19] [Ideas]
Currently, believe its only possible to show Charmstone with modifier keys held down (shift, function, cmd, etc), but if it can be triggered with a keyboard shortcut, that would enable 3rd party apps like KeyboardMaestro or BetterTouchTool (https://folivora.ai/) to tie Charmstone usage to mouse/trackpad gestures, or generally open it up to more customizable launch cases.

Currently BetterTouchTool does not allow modifier keys (alone) to be triggers for actions.  For it to function, there needs to be a secondary action tied to modifiers (ie - non-modifier keypress or an mouse-click)

**Comment by rxhanson** [comment]
Makes sense to me, I'll see if I can roll this into an upcoming release. 

There is one other way to trigger Charmstone, by using a mouse button. The setting is hidden and only enabled via terminal command, followed by an app restart:

`defaults write com.knollsoft.Charmstone reticleMouseButton -int 3`

Maybe that can get your desired behavior in the meantime.

**Comment by nikoloz110** [comment]
@hairybridget   hey,  did you manage to make it work?  I am using BetterTouchTool, but modifier + mouse click can not be sent to charmstone, only modifier + keypresses are allowed

**Reply by rxhanson** [reply]
There is now a URL api that you can use to trigger Charmstone from other apps, on the readme of this repo:
https://github.com/rxhanson/Charmstone-Community#using-urls-to-show-and-close-the-charmstone-window

**Reply by nikoloz110** [reply]
thanks, with that I can make charmstone window appear, but the window disappears the moment I move the cursor, hence I can not open apps :(

**Reply by nikoloz110** [reply]
@rxhanson  while looking through github, I saw another user had the same issue, (charmstone window disappearing, regardless how you triggered url, via alfred, bettertouchrool or simply via chrome),  could be a bug

**Reply by rxhanson** [reply]
I'll give this another look and see if I can get to the bottom of it.

**Reply by rxhanson** [reply]
Hmm. I still can't reproduce it. At a start, can you send me your config with this terminal command? (you can redact out the near/far selections if desired)

`defaults read com.knollsoft.Charmstone`

**Comment by tonyZazzle** [comment]
Looking for a similar feature led me here. I have a Logitech G603 mouse and am trying to configure that when I press and hold a mouse button it opens charmstone. The apps I'm hoping to trigger it from are either BetterMouse or BetterTouchTool, whichever works.

I've tried to use `defaults write com.knollsoft.Charmstone reticleMouseButton -int 10` which doesn’t seem to work

Also tried mapping the mouse button to `open - u charmstone://show` which didn’t work. Fyi, running that command, with and w/o the `-u` in the terminal also didn't work for me.

my settings:
```
defaults read com.knollsoft.Charmstone
{
    "NSStatusItem Preferred Position Item-0" = 1765;
    arrowKeysDisplays = 2;
    lastVersion = 31;
    launchOnLogin = 1;
    reticleDisplayMinDist = 5;
    reticleFlags = 0;
    reticleMouseButton = 10;
    safeAreaSize = 66;
}
```

Fyi, also a huge fan of Rectangle and was able to get windows throwing working using my mouse. Would love to use that same feature in Charmstone.

**Reply by rxhanson** [reply]
A couple of quick sanity checks:
1. Did you restart Charmstone after executing that terminal command?
2. Your open command has a space between `-` and `u`, but perhaps you had it entered correctly in Terminal? `open -u charmstone://show` works for me, so if it doesn't then maybe we can narrow down what's different on your Mac?

I'd like to finally get to the bottom of the the url not working for some people - let me know if you're up for using a debug build.
